src/odemis/odemisd/mdupdater.py

```diff
@@ -352,6 +352,11 @@ def observeFilter(self, filter, comp_affected):
         def updateOutWLRange(pos, fl=filter, comp_affected=comp_affected):
             wl_out = fl.axes["band"].choices[pos["band"]]
             comp_affected.updateMetadata({model.MD_OUT_WL: wl_out})
+            # apply lateral chromatic correction to align with the reference channel
+            apply_transform = fl.getMetadata().get(model.MD_CHROMATIC_COR, None)
+            fl_pos_band = fl.position.value["band"]
+            if apply_transform is not None and fl_pos_band in apply_transform.keys():
```

Comment:
You don't need `.keys()` to check a key is in a dict. Just write `fl_pos_band in apply_transform`, or maybe you can do it in a "greedy" way: directly read `apply_transform[fl_pos_band]`, and if it's not there, it'll raise a KeyError that you can catch.